modules/account-lib/test/unit/coin/dot/transactionBuilder/base.ts

```diff
@@ -178,4 +181,183 @@ xdescribe('Dot Transfer Builder', () => {
       should.deepEqual(builder.getTip(), undefined);
     });
   });
+
+  describe('add signature', () => {
+    const factory = register('tdot', TransactionBuilderFactory);
+
+    it('should add a signature to transaction', async () => {
+      const transferBuilder = factory
+        .getTransferBuilder()
+        .amount('90034235235322')
+        .sender({ address: sender.address })
+        .to({ address: receiver.address })
+        .validity({ firstValid: 3933, maxDuration: 64 })
+        .referenceBlock('0x149799bc9602cb5cf201f3425fb8d253b2d4e61fc119dcab3249f307f594754d')
+        .sequenceId({ name: 'Nonce', keyword: 'nonce', value: 200 })
+        .fee({ amount: 0, type: 'tip' });
+
+      transferBuilder.sign({ key: accounts.account1.secretKey });
+      const signedTx = await transferBuilder.build();
+      const signature = signedTx.signature[0];
+
+      // verify rebuilt transaction contains signature
+      const rawTransaction = signedTx.toBroadcastFormat() as string;
+      const rebuiltSignedTransaction = await factory
+        .from(rawTransaction)
+        .validity({ firstValid: 3933, maxDuration: 64 })
+        .referenceBlock('0x149799bc9602cb5cf201f3425fb8d253b2d4e61fc119dcab3249f307f594754d')
+        .build();
+      rebuiltSignedTransaction.signature.should.deepEqual(signedTx.signature);
+
+      const transferBuilder2 = factory
+        .getTransferBuilder()
+        .amount('90034235235322')
+        .sender({ address: sender.address })
+        .to({ address: receiver.address })
+        .validity({ firstValid: 3933, maxDuration: 64 })
+        .referenceBlock('0x149799bc9602cb5cf201f3425fb8d253b2d4e61fc119dcab3249f307f594754d')
+        .sequenceId({ name: 'Nonce', keyword: 'nonce', value: 200 })
+        .fee({ amount: 0, type: 'tip' });
+      transferBuilder2.addSignature({ pub: accounts.account1.publicKey }, Buffer.from(signature, 'hex'));
```

Comment:
Why not concatenate the signature addition to the previous builder step (`fee()`)?

Reply:
It looks like addSignature is consistent with the behavior of `sign()` where it returns void so it has it can't be concatenated like so because it will throw an error. Has to be standalone because of the return type.

```
.
.
.fee({ amount: 0, type: 'tip' })
.addSignature({ pub: accounts.account1.publicKey }, Buffer.from(signature, 'hex')).build;
```

Reply:
Sounds like a bug in `sign()` :/  anyway, outside this PR scope